src/prefect/server/api/flow_runs.py

```diff
@@ -520,27 +520,142 @@ async def set_flow_run_state(
     return orchestration_result
 
 
+async def _handle_paused_flow_run_input(
+    session, flow_run, run_input, flow_policy, orchestration_parameters
+):
+    state = flow_run.state
+    keyset = state.state_details.run_input_keyset
+
+    schema_json = await models.flow_run_input.read_flow_run_input(
+        session=session, flow_run_id=flow_run.id, key=keyset["schema"]
+    )
+
+    if schema_json is None:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=f"{state.type} flow run's input schema was not found",
+        )
+
+    try:
+        schema = orjson.loads(schema_json.value)
+    except orjson.JSONDecodeError:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=f"{state.type} flow run's input schema is not valid JSON",
+        )
+
+    try:
+        parsed_run_input = orjson.loads(run_input)
+    except orjson.JSONDecodeError as exc:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=f"Run input was not valid JSON: {exc.msg}",
+        )
+
+    try:
+        jsonschema.validate(parsed_run_input, schema)
+    except (jsonschema.ValidationError, jsonschema.SchemaError) as exc:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            detail=f"Run input did not match expected schema: {exc.message}",
+        )
+
+    if state.state_details.pause_reschedule:
+        orchestration_result = await models.flow_runs.set_flow_run_state(
+            session=session,
+            flow_run_id=flow_run.id,
+            state=schemas.states.Scheduled(
+                name="Resuming", scheduled_time=pendulum.now("UTC")
+            ),
+            flow_policy=flow_policy,
+            orchestration_parameters=orchestration_parameters,
+        )
+    else:
+        orchestration_result = await models.flow_runs.set_flow_run_state(
+            session=session,
+            flow_run_id=flow_run.id,
+            state=schemas.states.Running(),
+            flow_policy=flow_policy,
+            orchestration_parameters=orchestration_parameters,
+        )
+
+    if orchestration_result.status != schemas.responses.SetStateStatus.ACCEPT:
```

Comment:
I think it's a good idea for the `resume` and `input` endpoints to behave the same way, but now I'm curious what the differences are. It looks like the `resume` route provides a subset of the `input` behavior but are there any situation I'd want to keep using `resume`?

Reply:
The resume endpoint is an older one in use by the UI. It uses orchestration semantics, i.e. 200 with an orchestration result even on failure. I'd like to keep these separate, so e.g. we would use this input API internally in cases where we control the key by default, such as when`send_input` is called to send input to a paused/suspended flow.